src/keys.ts

```diff
@@ -444,12 +444,12 @@ function prepareSecretKey(
     return key;
   }
 
-  if (typeof key === 'string') {
+  if (typeof key === 'string' || key instanceof Uint8Array) {
```

Comment:
Can we fit the other `TypedArray` types in here?

Maybe just call `binaryLikeToArrayBuffer` instead of type conditionals, and see if it throws.